pkg/wharfapi/requestHandler.go

```diff
@@ -73,16 +73,24 @@ func redactTokenInURL(urlStr string) string {
 	return sanitized
 }
 
-func doRequest(from string, method string, URLStr string, body []byte, authHeader string) (*io.ReadCloser, error) {
-	var redactedURL = redactTokenInURL(URLStr)
+func doRequestNew(from string, method string, baseURL string, path string, q url.Values, body []byte, authHeader string) ([]byte, error) {
```

Comment:
There are a lot of parameters here just for constructing the URL. Suggest extracting this to a separate function instead and only passing an `*url.URL`.

Could even extract the `method`, `authHeader`, `body`, and the suggested `*url.URL` parameters to a separate request building function and then only take in the `from` and a `*http.Request` parameter.

Reply:
Cleaned it up a bit, please check again